### PR TITLE
104 story comments

### PR DESCRIPTION
**Pull request by tostistos**

### CODE REVIEW

**Review by AlekseyPanas — COMMENTED** [review]
Approved of core content, awaiting unit tests

**Review by bac423 — DISMISSED** [review]
Awesome job

**Review by AlekseyPanas — APPROVED** [review]

**Review by AlekseyPanas — APPROVED** [review]

File: src/main/java/usecases/comment_as_guest/CagGatewayComments.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ * Gateway for Comment As Guest use case, used by a repository storing guest comments
+ */
+public interface CagGatewayComments {
+
+    /**
+     * Method for implementing repository to add comment to the given story
+     * @param storyId the id of the story the comment was made under
+     * @param displayName the chosen display name of the guest who commented
+     * @param comment the comment the guest wrote
+     * @return true if adding the comment was successful
```
Comment (joel-mat-03) [inline src/main/java/usecases/comment_as_guest/CagGatewayComments.java:16]: the method returns response and not boolean so you should describe the response

File: src/test/java/usecases/comment_as_guest/CagInteractorTests.java [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+            this.commentAsGuestSuccess = commentAsGuestSuccess;
+        }
+
+        /**
+         * Method imitating repository's commentAsGuest with custom element based on
+         * commentAsGuestSuccess parameter.
+         * @param storyId requested storyId
+         * @param displayName proposed displayName of a guest
+         * @param comment proposed comment
+         * @return "success" when commentAsGuestSuccess is true, and "failure" otherwise
```
Comment (joel-mat-03) [inline src/test/java/usecases/comment_as_guest/CagInteractorTests.java:76]: i think this was originally implemented as method returning a string, so you would probably have to change it to 'return success or failure response' or something along those lines

File: src/main/java/usecases/get_story_comments/GscGatewayComments.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+/**
+ * Gateway for Get Story Comments use case, used by a repository to get all comments from a story
+ */
+public interface GscGatewayComments {
+
+    /**
+     * Gets all comments from the story matching the provided id
+     * @param storyId the id for the story comments should be pulled from
+     * @return a string array containing all the story comments, each one string
```
Comment (joel-mat-03) [inline src/main/java/usecases/get_story_comments/GscGatewayComments.java:15]: change this return description

File: src/test/java/usecases/get_story_comments/GscInteractorTests.java [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+        public CustomizableGscGatewayComments(boolean getAllCommentsSuccess) {
+            this.getAllCommentsSuccess = getAllCommentsSuccess;
+        }
+
+        /**
+         * Method imitating repository's getAllComments with custom element based on
+         * getAllCommentsSuccess parameter.
+         * @param storyId requested storyId
+         * @return "success" and predetermined output when commentAsGuestSuccess is true,
+         * and "failure" with null otherwise
```
Comment (joel-mat-03) [inline src/test/java/usecases/get_story_comments/GscInteractorTests.java:73]: once again, better to write 'success or failure response'. i am assuming this method was originally supposed to return a string

**Review by joel-mat-03 — CHANGES_REQUESTED** [review]

**Review by joel-mat-03 — APPROVED** [review]

**Review by joel-mat-03 — APPROVED** [review]

**Review by bac423 — APPROVED** [review]
Awesome job

**Pull request merged**